### PR TITLE
Creation of content view configuration

### PR DESCRIPTION
**Pull request by mtodor**

### CODE REVIEW

File: src/ConfigTemplateDefinition.php [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+   */
+  protected function getDynamicDefinitions($key) {
+    $dynamic_mapping_definitions = [];
+    foreach ($this->definition[$key] as $generate_per_field) {
+      $path = explode('.', $generate_per_field['name']);
+
+      $dynamic_mapping_definitions[] = [
+        'path' => $path,
+        'type' => $generate_per_field['type'],
+        'resource' => NestedArray::getValue($this->getSourceConfig(), $path),
```
Comment (dbosen) [inline src/ConfigTemplateDefinition.php]: NestedArray::getValue() expects the first value to be a reference to an array, but ::getSourceConfig() return an array. assigning the output of ::getSourceConfig() beforehand should fix this.

File: template/testsite_builder.config_template.views_node_content.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      source_field: 'title'
+    -
+      type: 'changed'
+      source_field: 'changed'
+    -
+      type: 'created'
+      source_field: 'created'
+    -
+      type: 'boolean'
+      source_field: 'status'
```
Comment (dbosen) [inline template/testsite_builder.config_template.views_node_content.yml]: All boolean fields do have 'Published / Unpublished' from the status field as display values now, but we want that to be True / False

File: src/ConfigTemplateMerge.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+   * @param array $config
+   *   The base configuration.
+   * @param array $path
+   *   The configuration merge path.
+   *
+   * @return array
+   *   Returns configuration with applied merge changes.
+   */
+  public function applyMerge(array $config, array $path) {
+    if ($this->tactic == static::SKIP) {
```
Comment (dbosen) [inline src/ConfigTemplateMerge.php]: Looks like these comparisons could all be strict "===" comparisons

File: src/Plugin/TestsiteBuilder/ConfigTemplateType/ViewBundlePath.php [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+ */
+class ViewBundlePath extends Generic {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfigChangesForBundle(string $bundle, $source_config) {
+    $path = explode('/', $source_config);
+
+    for ($i = 1; $i < count($path); $i++) {
```
Comment (dbosen) [inline src/Plugin/TestsiteBuilder/ConfigTemplateType/ViewBundlePath.php]: Yes, it is a micro-optimization, but count should be called outside the for loop :-)

**Review by dbosen — CHANGES_REQUESTED** [review]
Not related to this PR, but the README should document, that you have to do `GRANT SUPER on *.* to 'drupaluser'@'localhost'`

File: src/Plugin/TestsiteBuilder/ConfigTemplateType/ViewConfigCacheTag.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+   * The entity type manager service.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager) {
```
Comment (dbosen) [inline src/Plugin/TestsiteBuilder/ConfigTemplateType/ViewConfigCacheTag.php]: you could scrap the __constructor here and in ViewDependencyConfig.php and change the ::create() to 
```
   public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
    $plugin = parent::static($configuration, $plugin_id, $plugin_definition);
    $plugin->entityTypeManager = $container->get('entity_type.manager');
    return $plugin;
  } 
```

File: template/testsite_builder.config_template.views_node_content.yml [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+      name: 'display.page_1.display_options.path'
+  field_type_mapping:
+    - type: 'entity_reference'
+      match_rules:
+        storage_settings:
+          target_type: 'taxonomy_term'
+      source_field: 'field_channel'
+      custom_generation_types:
+        -
+          type: 'view_filter'
```
Comment (dbosen) [inline template/testsite_builder.config_template.views_node_content.yml]: Lets ghet rid of these inline config, and replace it with the fallback configuration files, that we talked about

File: src/Plugin/TestsiteBuilder/ConfigTemplateType/ViewFilter.php [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+   *   The bundle.
+   * @param string $field_name
+   *   The field name.
+   * @param mixed $config
+   *   The plugin configuration.
+   *
+   * @return mixed
+   *   Returns plugin configuration or FALSE.
+   */
+  protected function applyPlugConfiguration(string $entity_type, string $bundle, string $field_name, $config) {
```
Comment (dbosen) [inline src/Plugin/TestsiteBuilder/ConfigTemplateType/ViewFilter.php]: This smells like it should also be made into a plugin, as soon as there are more plugin_ids we might want to handle, this gets messy.

**Comment by mtodor** [comment]
@dbosen all review comments should be addressed.

**Review by dbosen — APPROVED** [review]
Let's do it. It works, and looks ok. We can do all we need for now. The small bugs, we still encounter do not matter ATM. It's more important to get that running.

**Pull request merged**